Mathlib/Topology/Instances/EReal.lean

```diff
@@ -214,65 +214,56 @@ lemma liminf_neg : liminf (- v) f = - limsup v f :=
-    ((eventually_lt_of_lt_liminf) b_v)) fun _ ab_x ↦ c_ab.trans (add_lt_add ab_x.1 ab_x.2)
+lemma le_limsup_add : (limsup u f) + (liminf v f) ≤ limsup (u + v) f :=
+  add_le_of_forall_lt fun _ a_u _ b_v ↦ (le_limsup_iff).2 fun _ c_ab ↦
+    (((frequently_lt_of_lt_limsup) a_u).and_eventually ((eventually_lt_of_lt_liminf) b_v)).mono
+    fun _ ab_x ↦ c_ab.trans (add_lt_add ab_x.1 ab_x.2)
 
-lemma liminf_add_le_limsup_add_liminf (h : limsup u f ≠ ⊥ ∨ liminf v f ≠ ⊤)
-    (h' : limsup u f ≠ ⊤ ∨ liminf v f ≠ ⊥) :
+lemma liminf_add_le (h : limsup u f ≠ ⊥ ∨ liminf v f ≠ ⊤) (h' : limsup u f ≠ ⊤ ∨ liminf v f ≠ ⊥) :
     liminf (u + v) f ≤ (limsup u f) + (liminf v f) :=
-  le_add_of_forall_le_add h h' fun _ a_u _ b_v ↦ (liminf_le_iff).2 fun _ c_ab ↦
-    Frequently.mono (Frequently.and_eventually ((frequently_lt_of_liminf_lt) b_v)
-    ((eventually_lt_of_limsup_lt) a_u)) fun _ ab_x ↦ (add_lt_add ab_x.2 ab_x.1).trans c_ab
+  le_add_of_forall_lt h h' fun _ a_u _ b_v ↦ (liminf_le_iff).2 fun _ c_ab ↦
+    (((frequently_lt_of_liminf_lt) b_v).and_eventually ((eventually_lt_of_limsup_lt) a_u)).mono
+    fun _ ab_x ↦ (add_lt_add ab_x.2 ab_x.1).trans c_ab
+
+@[deprecated (since := "2024-11-11")] alias add_liminf_le_liminf_add := le_liminf_add
+@[deprecated (since := "2024-11-11")] alias limsup_add_le_add_limsup := limsup_add_le
+@[deprecated (since := "2024-11-11")] alias limsup_add_liminf_le_limsup_add := le_limsup_add
+@[deprecated (since := "2024-11-11")] alias liminf_add_le_limsup_add_liminf := liminf_add_le
 
 variable {a b : EReal}
 
 lemma limsup_add_bot_of_ne_top (h : limsup u f = ⊥) (h' : limsup v f ≠ ⊤) :
     limsup (u + v) f = ⊥ := by
-  apply le_bot_iff.1 (le_trans (limsup_add_le_add_limsup (Or.inr h') _) _)
-  · rw [h]; exact Or.inl bot_ne_top
+  apply le_bot_iff.1 ((limsup_add_le (.inr h') _).trans _)
+  · rw [h]; exact .inl bot_ne_top
   · rw [h, bot_add]
 
 lemma limsup_add_le_of_le (ha : limsup u f < a) (hb : limsup v f ≤ b) :
     limsup (u + v) f ≤ a + b := by
-  rcases eq_top_or_lt_top b with (rfl | h)
+  rcases eq_top_or_lt_top b with rfl | h
   · rw [add_top_of_ne_bot ha.ne_bot]; exact le_top
-  · exact le_trans (limsup_add_le_add_limsup (Or.inr (lt_of_le_of_lt hb h).ne) (Or.inl ha.ne_top))
-      (add_le_add ha.le hb)
+  · exact (limsup_add_le (.inr (hb.trans_lt h).ne) (.inl ha.ne_top)).trans (add_le_add ha.le hb)
 
 lemma liminf_add_gt_of_gt (ha : a < liminf u f) (hb : b < liminf v f) :
     a + b < liminf (u + v) f :=
-  lt_of_lt_of_le (add_lt_add ha hb) add_liminf_le_liminf_add
+  (add_lt_add ha hb).trans_le le_liminf_add
 
 lemma liminf_add_top_of_ne_bot (h : liminf u f = ⊤) (h' : liminf v f ≠ ⊥) :
     liminf (u + v) f = ⊤ := by
-  apply top_le_iff.1 (le_trans _ (add_liminf_le_liminf_add))
+  apply top_le_iff.1 (le_trans _ le_liminf_add)
   rw [h, top_add_of_ne_bot h']
 
-lemma limsup_le_iff {b : EReal} : limsup u f ≤ b ↔ ∀ c : ℝ, b < c → ∀ᶠ a : α in f, u a ≤ c := by
```

Comment:
Unless I've simply missed it, this lemma is just being deleted. Why?

Reply:
This is a lemma I wrote in the first iteration of my proof of `EReal.add_limsup_le_limsup_add`. Since then, I've written and used `Filter.limsup_le_iff`. In comparison, `EReal.limsup_le_iff` is much less general and slightly hackish (the comparison with reals only was due to the complexity of the older proofs of `EReal.add_limsup_le_limsup_add`.).

There are some little advantages of `EReal.limsup_le_iff` over `Filter.limsup_le_iff`  (`c` can be chosen real, and the inequality at the end is not strict), but it is very, very niche and IMHO is not worth the place it takes in Mathlib. I think `Filter.limsup_le_iff` would do fine anytime now.

Reply:
I'm leaving the _awaiting-author_ tag for now so that this doesn't get merged before you can see this answer. If you are satisfied with it, you can remove the tag; otherwise, I'll modify the PR so that it suits you.

Reply:
@j-loreaux Are you OK with this explanation?

Reply:
That's fine, but please add a deprecation.